internal/cmd/providers.go

```diff
@@ -97,34 +120,22 @@ PROVIDER is a short unique name of the identity provider bieng added (eg. okta)
 
 			_, err = client.CreateProvider(&api.CreateProviderRequest{
 				Name:         args[0],
-				URL:          options.URL,
-				ClientID:     options.ClientID,
-				ClientSecret: options.ClientSecret,
+				URL:          opts.URL,
+				ClientID:     opts.ClientID,
+				ClientSecret: opts.ClientSecret,
 			})
 			if err != nil {
 				return err
 			}
 
 			fmt.Printf("Provider %s added\n", args[0])
-
 			return nil
 		},
 	}
 
-	cmd.Flags().String("url", "", "Base URL of the domain of the OIDC identity provider (eg. acme.okta.com)")
-	cmd.Flags().String("client-id", "", "OIDC client ID")
-	cmd.Flags().String("client-secret", "", "OIDC client secret")
-
-	if err := cmd.MarkFlagRequired("url"); err != nil {
-		panic("cannot set flag [--url] as required")
-	}
-	if err := cmd.MarkFlagRequired("client-id"); err != nil {
-		panic("cannot set flag [--client-id] as required")
-	}
-	if err := cmd.MarkFlagRequired("client-secret"); err != nil {
-		panic("cannot set flag [--client-secret] as required")
-	}
```

Comment:
Requiring flags this way does not allow us to set values from env vars. We could choose to not accept values from env vars. This was a problem when we used `parseOptions` as well.

The code to replace this (the `Validate` method above) is only 2 extra lines of code, and it allows us to customize the message. I think even if we don't use env vars we can ignore this feature of `cobra` and do it ourselves.